kato/kato-azure/src/main/groovy/com/netflix/spinnaker/kato/azure/deploy/description/templates/AzureLoadBalancerResourceTemplate.groovy

```diff
@@ -0,0 +1,437 @@
+    variables.append(addVarEntry(String.format(variableEntry, loadBalancerBackEndNameVar, backEndName) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, frontEndIPConfigNameVar, frontEndIPConfigName) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, dnsNameforLBIPVar, dnsName) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, subnetNameVar, subnetName) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, subnetPrefixVar, "10.0.0.0/24") + ","))
+    variables.append(addVarEntry(String.format(variableEntry, addressPrefixVar, "10.0.0.0/16") + ","))
+    variables.append(addVarEntry(String.format(variableEntry, nicNameVar, nic) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, ipConfigNameVar, ipConfigName) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, backEndIPConfigNameVar, ipConfigName) + ","))
+
+    // Add "id" variables
+    variables.append(addVarEntry(String.format(variableEntry, loadBalancerIDVar, String.format(resourceIdLookupString, networkLoadBalancerType, loadBalancerNameVar)) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, virtualNetworkIdVar, String.format(resourceIdLookupString, networkVNetType, virtualNetworkNameVar)) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, publicIPAddressIDVar, String.format(resourceIdLookupString, networkIPAddressesType, publicIPAddressNameVar)) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, frontEndIPConfigIDVar, getFrontEndIPConfigIDLookup()) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, subnetRefIdVar, getSubnetRefID()) + ","))
+    variables.append(addVarEntry(String.format(variableEntry, nicIdVar, String.format(resourceIdLookupString, networkNicType, nicNameVar))))
+
+    String.format(baseVariablesTemplate, variables.toString())
+  }
+
+  protected static String getParametersTemplate(UpsertAzureLoadBalancerDescription description) {
+    String.format(baseParametersTemplate, locationParameter)
+  }
+
+  protected static String getResourcesTemplate(UpsertAzureLoadBalancerDescription description) {
+    StringBuilder resources = new StringBuilder();
+    resources.append(getResourceEntry(getPublicIPAddressResource(), "," ))
+    resources.append(getResourceEntry(getVirtualNetworkResource(), ","))
+    resources.append(getResourceEntry(getNetworkInterfaceResource(), ","))
+    resources.append(getResourceEntry(getLoadBalancerResource(description),""))
+
+    String.format(baseResourceTemplate, resources.toString())
+  }
+
+  // get resource entries
+  private static String getLoadBalancerResource(UpsertAzureLoadBalancerDescription description) {
+    StringBuilder loadBalancers = new StringBuilder()
+    loadBalancers.append(String.format(resourceHeader, apiVersion, String.format(variableString, loadBalancerNameVar), networkLoadBalancerType))
```

Comment:
yes, it generates JSON. Going forward we expect to make a call to a general service which given a list of descriptions for the cloud resources to be created/updated it will return a full JSON template. I believe Haishi mentioned that (cc @HaishiBai )